apix-impl/src/test/java/eu/xenit/apix/tests/search/FtsNodeVisitorTest.java

```diff
@@ -0,0 +1,96 @@
+import eu.xenit.apix.alfresco.search.FtsNodeVisitor;
+import eu.xenit.apix.data.QName;
+import eu.xenit.apix.properties.PropertyDefinition;
+import eu.xenit.apix.search.QueryBuilder;
+import eu.xenit.apix.search.nodes.SearchSyntaxNode;
+import eu.xenit.apix.search.visitors.SearchSyntaxPrinter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class FtsNodeVisitorTest {
+
+    private final static Logger logger = LoggerFactory.getLogger(FtsNodeVisitorTest.class);
+
+    private Map<String, String> propertyToDataType = new HashMap<String, String>() {{
+        put("{tenant.model}stringProperty1", "d:text");
+        put("{tenant.model}stringProperty2", "d:text");
+        put("{tenant.model}intProperty", "d:int");
+    }};
+
+    @Test
+    public void testIntTypeInvalid() {
+        SearchSyntaxNode querySyntaxTree = generateAllQuery("5500012345");
+        PropertyService propertyService = generatePropertyServiceMock();
+
+        String ftsQuery = toFts(querySyntaxTree, propertyService);
+        assertThat("Fts search String does not contain wanted term", ftsQuery,
+                containsString("{tenant.model}stringProperty1"));
+        assertThat("Fts search String does not contain wanted term", ftsQuery,
+                containsString("{tenant.model}stringProperty2"));
+        assertThat("Fts search String contains term that shoud have been filtered out", ftsQuery,
+                not(containsString("{tenant.model}intProperty")));
+    }
+
+    @Test
+    public void testIntTypeValid() {
+        SearchSyntaxNode querySyntaxTree = generateAllQuery("1500012345");
+        PropertyService propertyService = generatePropertyServiceMock();
+
+        String ftsQuery = toFts(querySyntaxTree, propertyService);
+        assertThat("Fts search String does not contain wanted term", ftsQuery,
+                containsString("{tenant.model}stringProperty1"));
+        assertThat("Fts search String does not contain wanted term", ftsQuery,
+                containsString("{tenant.model}stringProperty2"));
+        assertThat("Fts search String does not contain wanted term", ftsQuery,
+                containsString("{tenant.model}intProperty"));
+    }
+
+    private SearchSyntaxNode generateAllQuery(String value) {
+        QueryBuilder querySyntaxTree = new QueryBuilder()
+                .startAnd()
+                .startOr();
+        for (String property : propertyToDataType.keySet()) {
+            querySyntaxTree = querySyntaxTree.property(property, value);
+        }
+        return querySyntaxTree
+                .end()
+                .term("type", "{http://www.alfresco.org/model/content/1.0}content")
+                .end()
+                .create();
+    }
+
+    private PropertyService generatePropertyServiceMock() {
+        PropertyService propertyService = mock(PropertyService.class);
```

Comment:
@driesdeproost PropertyService is an service of Alfred API so you don't need to mock it. It is better to create a stub like Toon said and then mock the Alfresco services you need. In this case you will need a mock for the dictionary service I believe.